src/System.Private.CoreLib/shared/System/Runtime/Loader/AssemblyLoadContext.cs

```diff
@@ -100,6 +100,8 @@ private protected AssemblyLoadContext(bool representsTPALoadContext, bool isColl
                 _id = s_nextId++;
                 s_allContexts.Add(_id, new WeakReference<AssemblyLoadContext>(this, true));
             }
+
+            Resolving += ResolveSatelliteAssembly;
```

Comment:
It does not sound right that we are always unconditionally subscribing to event. Should this be a callback from the runtime instead?